sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2764,6 +2764,14 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val DYNAMIC_DECIDE_BUCKETING_ENABLED =
+    buildConf("spark.sql.sources.dynamic.decide.bucketing.enabled")
```

Comment:
How about adding this param under `spark.sql.sources.bucketing.xxx`? `spark.sql.sources.bucketing.enableAutoBucketScan` or something?

Reply:
+1

Reply:
or `dynamicBucketScan`